File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Calling WCF Services from Timer Jobs/VB/WCF_CallServiceFromTimerJob/Features/Feature1/Feature1.EventReceiver.vb
Option Explicit On
Option Strict On

Imports System
Imports System.Runtime.InteropServices
Imports System.Security.Permissions
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.Administration
Imports Microsoft.SharePoint.Security

''' <summary>
''' This class handles events raised during feature activation, deactivation, installation, uninstallation, and upgrade.
''' </summary>
''' <remarks>
''' The GUID attached to this class may be used during packaging and should not be modified.
''' </remarks>

<GuidAttribute("1e060bdb-133e-44d6-a25f-04495afbb2a4")> _
Public Class Feature1EventReceiver 
    Inherits SPFeatureReceiver

    Const TIMER_JOB_NAME As String = "WCFCaller"

    Public Overrides Sub FeatureActivated(ByVal properties As SPFeatureReceiverProperties)
        'In this event we create and install the timer job
        'Start by finding the SPSite.
        Dim site As SPSite = DirectCast(properties.Feature.Parent, SPSite)
        'Make sure the timer job isn't already registered
        For Each job As SPJobDefinition In site.WebApplication.JobDefinitions
            If job.Name = TIMER_JOB_NAME Then
                job.Delete()
            End If
        Next
        'Create a configure the new Timer job
        Dim wcfTimerJob As CallingTimerJob = New CallingTimerJob(TIMER_JOB_NAME, site.WebApplication)
        Dim jobSchedule As SPMinuteSchedule = New SPMinuteSchedule()
        jobSchedule.BeginSecond = 0
        jobSchedule.EndSecond = 59
        jobSchedule.Interval = 5
        wcfTimerJob.Schedule = jobSchedule
        wcfTimerJob.Update()
    End Sub


    Public Overrides Sub FeatureDeactivating(ByVal properties As SPFeatureReceiverProperties)
        Dim site As SPSite = DirectCast(properties.Feature.Parent, SPSite)
        'Make sure the timer job isn't already registered
        For Each job As SPJobDefinition In site.WebApplication.JobDefinitions
            If job.Name = TIMER_JOB_NAME Then
                job.Delete()
            End If
        Next
    End Sub

End Class

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Calling WCF Services from Timer Jobs/C#/WCF_CallServiceFromTimerJob/Features/Feature1/Feature1.EventReceiver.cs
using System;
using System.Runtime.InteropServices;
using System.Security.Permissions;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Administration;
using Microsoft.SharePoint.Security;

namespace WCF_CallServiceFromTimerJob.Features.Feature1
{
    /// <summary>
    /// This class handles events raised during feature activation, deactivation, installation, uninstallation, and upgrade.
    /// </summary>
    /// <remarks>
    /// The GUID attached to this class may be used during packaging and should not be modified.
    /// </remarks>

    [Guid("683e7fb9-f5ab-44e7-ad1c-4bdbf0338352")]
    public class Feature1EventReceiver : SPFeatureReceiver
    {
        const string TIMER_JOB_NAME = "WCFCaller";

        public override void FeatureActivated(SPFeatureReceiverProperties properties)
        {
            //In this event we create and install the timer job
            //Start by finding the SPSite.
            SPSite site = properties.Feature.Parent as SPSite;
            //Make sure the timer job isn't already registered
            foreach(SPJobDefinition job in site.WebApplication.JobDefinitions)
            {
                if(job.Name == TIMER_JOB_NAME)
                {
                    job.Delete();
                }
            }
            //Create a configure the new Timer job
            WCF_CallingTimerJob wcfTimerJob = new WCF_CallingTimerJob(TIMER_JOB_NAME, site.WebApplication);
            SPMinuteSchedule jobSchedule = new SPMinuteSchedule();
            jobSchedule.BeginSecond = 0;
            jobSchedule.EndSecond = 59;
            jobSchedule.Interval = 5;
            wcfTimerJob.Schedule = jobSchedule;
            wcfTimerJob.Update();
        }

        public override void FeatureDeactivating(SPFeatureReceiverProperties properties)
        {
            //In this event we must clean up by deleting the timer job
            SPSite site = properties.Feature.Parent as SPSite;
            //Locate the right timer job
            foreach (SPJobDefinition job in site.WebApplication.JobDefinitions)
            {
                if (job.Name == TIMER_JOB_NAME)
                {
                    //This one is the right job. Delete it.
                    job.Delete();
                }
            }
        }

    }
}
